logtail/filch/filch.go

```diff
@@ -91,6 +93,17 @@ func (f *Filch) scan() ([]byte, error) {
 func (f *Filch) Write(b []byte) (int, error) {
 	f.mu.Lock()
 	defer f.mu.Unlock()
+	fi, err := f.cur.Stat()
```

Comment:
it would be nice to avoid the stat on every write. put a log counter in Filch, n and check every 100 writes?

Reply:
Or go stateless and do checks randomly. Or put a size counter in Filch, at which point you don't need a stat ever.